---
title: "Overview of the iSEE package"
author: "Kevin Rue-Albrecht, Charlotte Soneson, Federico Marini, and Aaron Lun"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    toc_float: true
    toc_depth: 3
    fig_width: 5
    number_sections: true
bibliography: "`r iSEEWorkshopEuroBioc2020::get_bib_file()`"
vignette: >
  %\VignetteIndexEntry{Overview of the iSEE package}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  cache = FALSE,
  collapse = TRUE,
  comment = "#>",
  crop = NA
)
suppressPackageStartupMessages({
    require(BiocStyle)
})
htmltools::tagList(rmarkdown::html_dependency_font_awesome())
```

```{r, eval=!exists("SCREENSHOT"), include=FALSE}
SCREENSHOT <- function(x, ...) knitr::include_graphics(x)
```

# Introduction

This vignette provides an overview of the graphical interface of `r Biocpkg("iSEE")` applications.
To follow along, make sure that you have launched an `r Biocpkg("iSEE")` instance as described in the code block at the end of the previous section.
Note that in the default configuration, the panels do not look exactly like the ones shown in the screenshots below.
For example, data points are not immediately colored, and the default annotation variables displayed by each panel may differ.
Later sections of this workshop demonstrate how to modify the content of the panels and how they are displayed. 

Note that for simplicity, we typically refer to a `SummarizedExperiment` in this workshop; however, `r Biocpkg("iSEE")` works seamlessly for objects of any class extending `SummarizedExperiment` as well (e.g., `SingleCellExperiment`, `DESeqDataSet`).
That said, some types of panels -- such as the `Reduced dimension plot` -- are only available for objects that contain a `reducedDim` slot (in particular, `SingleCellExperiment` objects); the basic `SummarizedExperiment` class does not contain this slot.
In this workshop, we refer to the rows of the `SummarizedExperiment` object as 'features' (these can be genes, transcripts, genomic regions, etc) and to the columns as 'samples' (which, in our example data set, are single cells).

# The default iSEE app

Using the demonstration data set, we can launch an `r Biocpkg("iSEE")` instance for exploring this data set using the `iSEE()` function without any further argument.
This will produce an app using the default configuration; that is, the app instance will include one panel of each built-in class for which the relevant information is available in the `SummarizedExperiment` object.

```{r, message=FALSE}
library(iSEE)
library(iSEEWorkshopEuroBioc2020)
sce <- load_demo_sce("pbmc3k")
app <- iSEE(sce)
```

```{r runApp, eval=FALSE}
shiny::runApp(app)
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-initial.png", delay=20)
```

The main argument to the `iSEE()` function is a `SummarizedExperiment` object, or an object of any class extending `SummarizedExperiment` (such as `SingleCellExperiment`, in this case).
No other restrictions are made on the type of data stored in the object, and `r Biocpkg("iSEE")` can be used for interactive visualization of many different types of data.
It is also worth noting that for various types of data, Bioconductor packages provides functionality for directly importing quantifications generated by external software packages into a `SummarizedExperiment` object.
For example, the `r Biocpkg("DropletUtils")` package can read quantifications from the 10x Genomics CellRanger pipeline for single-cell RNA-seq data, and the `r Biocpkg("tximeta")` package can be used to read data from transcript quantification pipelines into a `SummarizedExperiment` object.

While we will not make explicit use of it in this workshop, we note that it is also possible to call `iSEE()` without providing a `SummarizedExperiment` object. 
In that case, the user will be prompted to upload such an object, serialized into a `.rds` file. 
It is also possible to import a specification of the initial panel setup. 

```{r}
app <- iSEE()
```

```{r, eval=FALSE}
shiny::runApp(app)
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-no-se.png", delay=10)
```

# The built-in panel types {#user-interface}

The `r Biocpkg("iSEE")` user interface consists of a number of panels, each displaying the data provided in the `SummarizedExperiment` from a specific perspective.
There are 8 standard panel types; 6 plot panels and 2 table panels, all showcased in the figure shown at the end of section [Preparation of example scRNAseq data](#prepare-scrnaseq-data).
In the default configuration, one panel of each type is included when launching the `r Biocpkg("iSEE")` user interface.
However, users are free to rearrange, resize, remove or add panels freely, as described in the later section [Configuration of the app interface](#configuration-ui).
We provide a brief overview of each panel type in the following subsections.

In addition to the 8 standard panel types included in `r Biocpkg("iSEE")`, users can create custom panels (both plots and tables).
Moreover, the `r Biocpkg("iSEEu")` (iSEE universe) package contains additional panel types. 
The creation and configuration of custom panels is discussed later, in the dedicated section [Custom panels](#custom-panels).

<!--
This is perhaps not very pedagogical - the app will not look like the one below if it is just opened with the object created above.
But it's nice to show the capabilities of the panels too, not just the default settings.
I added a few sentences to the introduction of this section to try to address this.
--> 

## Reduced dimension plot {#reddimplot}

The reduced dimension plot can display any reduced dimension representation that is present in the `reducedDim` slot of the `SingleCellExperiment` object.

Note that this slot is not defined for the base `SummarizedExperiment` class, in which case the user interface does not allow the inclusion of panels of this type.
 
```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ReducedDimensionPlot(
  PanelWidth = 12L, 
  Type = "TSNE", ColorBy = "Column data", 
  ColorByColumnData = "Cluster"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-reduceddimensionplot.png", delay=10)
```

## Column data plot {#coldataplot}

The column data plot can display one or two of the provided column annotations (from the `colData` slot).
Depending on the class of the selected annotations, the panel shows either a scatter plot, a violin plot, or a Hinton diagram [@Hinton1991-hintondiagram; @Bremner1994-hintonplots].

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ColumnDataPlot(
  PanelWidth = 12L, 
  XAxis = "Column data", XAxisColumnData = "Cluster",
  YAxis = "log10_total"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-columndataplot.png", delay=10)
```

## Row data plot {#rowdataplot}

Analogous to the [column data plot](#coldataplot) above, the row data plot displays one or two of the provided row annotations (from the `rowData` slot).
Depending on the class of the selected annotations, the panel displays either a scatter plot, a violin plot, or a Hinton plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(RowDataPlot(
  PanelWidth = 12L, XAxis = "Row data",
  YAxis = "detected", XAxisRowData = "log10_total"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-rowdataplot.png", delay=10)
```

## Complex heatmap {#heatmap}

The complex heatmap panel displays, for any assay, the observed values for a subset of the features across the samples.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ComplexHeatmapPlot(
  PanelWidth = 12L, Assay = "logcounts", CustomRows = TRUE,
  CustomRowsText = c("CD14", "CD27", "CD52", "CD68", "CD19", "CD4"),
  ColumnData = "Cluster", AssayCenterRows = TRUE
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-complexheatmapplot.png", delay=10)
```

## Feature assay plot {#featassayplot}

The feature assay plot displays the observed values for one feature across the samples.
It is also possible to plot the observed values for two features, in a scatter plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(FeatureAssayPlot(
  PanelWidth = 12L, XAxis = "Column data",
  XAxisColumnData = "Cluster", YAxisFeatureName = "CD27",
  Assay = "logcounts", ColorBy = "Column data",
  ColorByColumnData = "Cluster"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-featureassayplot.png", delay=10)
```

## Sample assay plot {#sampassayplot}

Analogous to the [feature assay plot](#featassayplot) above, the sample assay plot shows the observed values for all features, for one of the samples.
It is also possible to plot the observed values for two samples, in a scatter plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(SampleAssayPlot(
  PanelWidth = 12L, XAxis = "Sample name",
  XAxisSampleName = "Cell1", YAxisSampleName = "Cell2",
  PointSize = 2
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-sampleassayplot.png", delay=10)
```

## Row data table {#rowdatatable}

The row data table displays all information provided in the `rowData` slot of the `SummarizedExperiment` object, leveraging the interactivity provided by the `r CRANpkg("DT")` package.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(RowDataTable(PanelWidth = 12L)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-rowdatatable.png", delay=10)
```

## Column data table {#coldatatable}

Analogous to the [row data table](#rowdatatable) above, the column data table displays all information provided in the `colData` slot of the `SummarizedExperiment` object.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ColumnDataTable(PanelWidth = 12L)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-columndatatable.png", delay=10)
```

# Collapsible boxes with display controls {#collapsible-boxes}

## 'Data parameters' collapsible box {#data-parameters-box}

Each plot panel type has a `Data parameters` collapsible box.
This box has different content for each panel type, but in all cases it lets the user control the data that is displayed in the plot.

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ColumnDataPlot(
  PanelWidth = 12L, DataBoxOpen = TRUE,
  XAxis = "Column data", XAxisColumnData = "Cluster",
  YAxis = "log10_total"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-columndataplot-dataparameters.png", delay=10)
```

## 'Visual parameters' collapsible box {#visual-parameters-box}

In contrast to the `Data parameters` collapsible box that lets users control _what_ is displayed in the plot, the `Visual parameters` box lets users control _how_ the information is displayed.

This collapsible box contains the controls to change the size, shape, opacity, and color of the points, to facet the plot by any available categorical annotation, to subsample points for increased speed of plot rendering, and to control how legends are displayed.

```{r, message=FALSE, echo=FALSE}
app <- iSEE(sce, initial = list(ColumnDataPlot(
  PanelWidth = 12L, VisualBoxOpen = TRUE,
  XAxis = "Column data", XAxisColumnData = "Cluster",
  YAxis = "log10_total", 
  VisualChoices = c("Color", "Point", "Facet", "Shape", "Size", "Text")
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-columndataplot-visualparameters.png", delay=10)
```

## 'Selection parameters' collapsible box {#selection-parameter-box}

The `Selection parameters` collapsible box provides controls to transfer selections of points (features or samples) between panels.

We discuss point transmission in more detail [later in this workshop](#transmit-point-selections).

```{r, message=FALSE, echo = FALSE}
app <- iSEE(sce, initial = list(ReducedDimensionPlot(
  PanelWidth = 12L, SelectionBoxOpen = TRUE,
  Type = "TSNE", ColorBy = "Column data", 
  ColorByColumnData = "Cluster"
)))
```

```{r, echo=FALSE, out.width='100%'}
SCREENSHOT("screenshots/isee-reduceddimensionplot-selectionparameters.png", delay=10)
```

# Additional controls {#additional-controls}

At the top-right corner of the iSEE application, users can find additional controls for reproducibility, configuration, and help.
Each of these is discussed in more detail later in this workshop.
Links are provided in the last column of the table below.

```{r, echo = FALSE, eval = TRUE}
panel_organization_fig <- "controls/PanelOrganization.png"
export_fig <- "controls/Export.png"
documentation_fig <- "controls/Documentation.png"
additional_info_fig <- "controls/AdditionalInfo.png"
```

|              |            |                                        |
|--------------|:----------:|----------------------------------------|
| Organization | `r knitr::include_graphics(panel_organization_fig, dpi = 180)` | [Organize panels](#configuration-ui) <br> [Examine panel chart](#transmit-point-selections) |
| Export        | `r knitr::include_graphics(export_fig, dpi = 180)`       | [Download panel output](#download-output) <br> [Extract R code](#reproducibility) <br> [Display panel settings](#configuration-ui) |
| Documentation      | `r knitr::include_graphics(documentation_fig, dpi = 180)`     | [Quick tour](#sharing-tours) <br> [Open the vignette](#more-resources) |
| Additional info    | `r knitr::include_graphics(additional_info_fig, dpi = 180)`    | [About this session](#reproducibility) <br> [About iSEE](#more-resources) |
