<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Preprocessing a data set for iSEE • iSEEWorkshopEuroBioc2020</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Preprocessing a data set for iSEE">
<meta property="og:description" content="iSEEWorkshopEuroBioc2020">
<meta property="og:image" content="https://iSEE.github.io/iSEEWorkshopEuroBioc2020/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEWorkshopEuroBioc2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/dataset.html">Preprocessing a data set for iSEE</a>
    </li>
    <li>
      <a href="../articles/exercises.html">Hands-on workshop exercises</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/iSEE/iSEEWorkshopEuroBioc2020">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="dataset_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="dataset_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="dataset_files/anchor-sections-1.0/anchor-sections.js"></script><link href="dataset_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="dataset_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Preprocessing a data set for iSEE</h1>
                        <h4 class="author">Kevin Rue-Albrecht, Charlotte Soneson, Federico Marini, and Aaron Lun</h4>
            
            <h4 class="date">December 04, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iSEEWorkshopEuroBioc2020/blob/master/vignettes/dataset.Rmd"><code>vignettes/dataset.Rmd</code></a></small>
      <div class="hidden name"><code>dataset.Rmd</code></div>

    </div>

    
    
<div id="prepare-scrnaseq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-scrnaseq-data" class="anchor"></a>Preparation of example scRNAseq data</h1>
<p>In this workshop, we use example data from the <em><a href="https://bioconductor.org/packages/3.13/TENxPBMCData">TENxPBMCData</a></em> package. This package provides an <em>R / Bioconductor</em> resource for representing and manipulating different single-cell RNA-seq data sets profiling peripheral blood mononuclear cells (PBMC) generated by 10x Genomics (<a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/datasets</a>).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TENxPBMCData</span><span class="op">)</span></code></pre></div>
<p>The man page for the <code><a href="https://rdrr.io/pkg/TENxPBMCData/man/TENxPBMCData.html">TENxPBMCData()</a></code> function gives an idea of the datasets that are available from this package. It can be opened with the following command.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span><span class="op">(</span><span class="va">TENxPBMCData</span><span class="op">)</span></code></pre></div>
<p>Here, we use the <code>pbmc3k</code> dataset, which contains gene expression profiles for 2,700 single peripheral blood mononuclear cells. The first time this dataset is loaded, this command downloads the dataset to a local cache, which takes some time, depending on the speed of your internet connection. Subsequent times, the same command loads the dataset directly from the local cache.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TENxPBMCData/man/TENxPBMCData.html">TENxPBMCData</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"pbmc3k"</span><span class="op">)</span></code></pre></div>
<p>At this point we can inspect the dataset in the console.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 32738 2700 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(32738): ENSG00000243485 ENSG00000237613 ... ENSG00000215616</span>
<span class="co">#&gt;   ENSG00000215611</span>
<span class="co">#&gt; rowData names(3): ENSEMBL_ID Symbol_TENx Symbol</span>
<span class="co">#&gt; colnames: NULL</span>
<span class="co">#&gt; colData names(11): Sample Barcode ... Individual Date_published</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(0):</span></code></pre></div>
<p>The dataset is provided as an object of the <code>SingleCellExperiment</code> class. In particular, this summary view indicates that the following pieces of information are available:</p>
<ul>
<li>An assay matrix named <code>counts</code>
</li>
<li>Row (i.e., gene) names are Ensembl gene IDs</li>
<li>Row metadata include for each gene the official gene symbol, and the gene symbol used by the 10x CellRanger quantification pipeline</li>
<li>Column (i.e., cell) IDs are not initialized and left to <code>NULL</code>
</li>
<li>Column metadata include diverse information for each cell, including the cell barcode (<code>Barcode</code>) and the donor identifier (<code>Individual</code>).</li>
</ul>
<p>Note that a <code>SingleCellExperiment</code> object (or, more generally, any <code>SummarizedExperiment</code> object) like this one already contains sufficient information to launch an interactive application instance to visualize the available data and metadata, using the <code>iSEE()</code> function.</p>
<p>For the purpose of this workshop, we first apply some preprocessing to the <code>SingleCellExperiment</code> object, in order to populate it with more information that can be visualized with <code>iSEE</code>.</p>
<p>We start by adding column names to the object, and use gene symbols instead of Ensembl IDs as row names. In the case where multiple Ensembl identifiers correspond to the same gene symbol, the <code><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html">scater::uniquifyFeatureNames</a></code> function concatenates the Ensembl ID and the gene symbol in order to generate unique feature names.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Cell"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html">uniquifyFeatureNames</a></span><span class="op">(</span>
    ID <span class="op">=</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">ENSEMBL_ID</span>,
    names <span class="op">=</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol_TENx</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "MIR1302-10"   "FAM138A"      "OR4F5"        "RP11-34P13.7" "RP11-34P13.8"</span>
<span class="co">#&gt; [6] "AL627309.1"</span></code></pre></div>
<p>Next, we use the <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> package to calculate gene- and cell-level quality metrics. These metrics are added as columns to the <code>rowData</code> and <code>colData</code> slots of the <code>SingleCellExperiment</code> object, respectively. We also add some additional metrics that are not automatically computed by the <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> package.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"^MT-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQC.html">addPerCellQC</a></span><span class="op">(</span><span class="va">sce</span>, subsets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>MT <span class="op">=</span> <span class="va">MT</span><span class="op">)</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQC.html">addPerFeatureQC</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="va">sce</span><span class="op">$</span><span class="va">log10_total</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">total</span><span class="op">)</span>
<span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">n_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">detected</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span>
<span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">log10_total</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">sce</span>, <span class="st">"counts"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">sce</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 32738 2700 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span>
<span class="co">#&gt; rowData names(7): ENSEMBL_ID Symbol_TENx ... n_cells log10_total</span>
<span class="co">#&gt; colnames(2700): Cell1 Cell2 ... Cell2699 Cell2700</span>
<span class="co">#&gt; colData names(18): Sample Barcode ... total log10_total</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(0):</span></code></pre></div>
<p>We filter out a few cells with a large fraction of the counts coming from mitochondrial genes, since these may be damaged cells. Notice the reduced number of columns in the dataset below.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>, <span class="va">sce</span><span class="op">$</span><span class="va">subsets_MT_percent</span> <span class="op">&lt;</span> <span class="fl">5</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 32738 2643 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span>
<span class="co">#&gt; rowData names(7): ENSEMBL_ID Symbol_TENx ... n_cells log10_total</span>
<span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span>
<span class="co">#&gt; colData names(18): Sample Barcode ... total log10_total</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(0):</span></code></pre></div>
<p>Next, we calculate size factors and normalized and log-transformed expression values, using the <em><a href="https://bioconductor.org/packages/3.13/scran">scran</a></em> and <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> packages. Note that it is typically recommended to pre-cluster the cells before computing the size factors, as follows:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set.seed(1000)</span>
<span class="co"># clusters &lt;- scran::quickCluster(sce, BSPARAM = IrlbaParam())</span>
<span class="co"># sce &lt;- scran::computeSumFactors(sce, cluster = clusters, min.mean = 0.1)</span></code></pre></div>
<p>However, for time reasons, we will skip the pre-clustering step in this workshop.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">sce</span>, <span class="st">"counts"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">sce</span>, <span class="st">"counts"</span><span class="op">)</span>, <span class="st">"sparseMatrix"</span><span class="op">)</span> <span class="co"># dirty fix for DockerHub</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/computeSumFactors.html">computeSumFactors</a></span><span class="op">(</span><span class="va">sce</span>, min.mean <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/sizeFactors.html">sizeFactors</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;  0.1164  0.7309  0.9004  1.0000  1.1635  9.8580</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></code></pre></div>
<p>In order to extract the most informative genes, we first model the mean-variance trend and decompose the variance into biological and technical components.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dec</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="va">top.dec</span> <span class="op">&lt;-</span> <span class="va">dec</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">dec</span><span class="op">$</span><span class="va">bio</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="op">]</span> 
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">top.dec</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 6 columns</span>
<span class="co">#&gt;              mean     total      tech       bio      p.value          FDR</span>
<span class="co">#&gt;         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="co">#&gt; LYZ       1.63950   3.92209  0.884955   3.03713 4.59558e-117 2.54472e-113</span>
<span class="co">#&gt; S100A9    1.07323   3.36899  0.778205   2.59079 2.72222e-110 9.04432e-107</span>
<span class="co">#&gt; HLA-DRA   1.56937   3.16577  0.880146   2.28562  5.73029e-68  7.93263e-65</span>
<span class="co">#&gt; FTL       3.67846   2.88499  0.708226   2.17676  2.46520e-94  5.85028e-91</span>
<span class="co">#&gt; CD74      2.23653   2.75839  0.854474   1.90392  1.34728e-50  1.01732e-47</span>
<span class="co">#&gt; CST3      1.10995   2.59172  0.790231   1.80149  7.24671e-53  6.01912e-50</span></code></pre></div>
<p>Next, we apply Principal Components Analysis (PCA) and t-distributed Stochastic Neighbor Embedding (t-SNE) to generate low-dimensional representations of the cells in our data set. These low-dimensional representations are added to the <code>reducedDim</code> slot of the <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/BiocSingular">BiocSingular</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/denoisePCA.html">denoisePCA</a></span><span class="op">(</span><span class="va">sce</span>, technical <span class="op">=</span> <span class="va">dec</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"PCA"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 5</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html">runTSNE</a></span><span class="op">(</span><span class="va">sce</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span>, perplexity <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">sce</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span>
<span class="va">sce</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 32738 2643 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span>
<span class="co">#&gt; rowData names(7): ENSEMBL_ID Symbol_TENx ... n_cells log10_total</span>
<span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span>
<span class="co">#&gt; colData names(19): Sample Barcode ... log10_total sizeFactor</span>
<span class="co">#&gt; reducedDimNames(3): PCA TSNE UMAP</span>
<span class="co">#&gt; altExpNames(0):</span></code></pre></div>
<p>After this, we cluster the cells using a graph-based algorithm, and find ‘marker genes’ for each cluster as the genes that are significantly upregulated in the cluster compared to each of the other inferred clusters. The adjusted p-values from this test, for each cluster, are added to the <code>rowData</code> slot of the object.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snn.gr</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html">buildSNNGraph</a></span><span class="op">(</span><span class="va">sce</span>, use.dimred <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span>
<span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_walktrap.html">cluster_walktrap</a></span><span class="op">(</span><span class="va">snn.gr</span><span class="op">)</span>
<span class="va">sce</span><span class="op">$</span><span class="va">Cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">$</span><span class="va">membership</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">Cluster</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13  14 </span>
<span class="co">#&gt; 143 330 212 362 224 389 338 144 142 129  79 115  25  11</span>
<span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/findMarkers.html">findMarkers</a></span><span class="op">(</span><span class="va">sce</span>, groups <span class="op">=</span> <span class="va">sce</span><span class="op">$</span><span class="va">Cluster</span>,
                              test.type <span class="op">=</span> <span class="st">"t"</span>,
                              direction <span class="op">=</span> <span class="st">"up"</span>, pval.type <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">markers</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"FDR_cluster"</span>, <span class="va">i</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> 
        <span class="va">markers</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">FDR</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, 
                               <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">markers</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="op">}</span>
<span class="va">sce</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 32738 2643 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(32738): MIR1302-10 FAM138A ... AC002321.2 AC002321.1</span>
<span class="co">#&gt; rowData names(21): ENSEMBL_ID Symbol_TENx ... FDR_cluster13</span>
<span class="co">#&gt;   FDR_cluster14</span>
<span class="co">#&gt; colnames(2643): Cell1 Cell2 ... Cell2699 Cell2700</span>
<span class="co">#&gt; colData names(20): Sample Barcode ... sizeFactor Cluster</span>
<span class="co">#&gt; reducedDimNames(3): PCA TSNE UMAP</span>
<span class="co">#&gt; altExpNames(0):</span></code></pre></div>
<p>Finally, we assign a label to each cell, based on their individual transcriptome profiles, using the <code>SingleR</code> method and the Monaco immune data (<a href="https://doi.org/10.1016/j.celrep.2019.01.041" class="uri">https://doi.org/10.1016/j.celrep.2019.01.041</a>) as a reference.<br>
For each prediction, we assign the labels values to a specific <code>colData</code> element</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/SingleR">SingleR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/celldex">celldex</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'celldex'</span>
<span class="co">#&gt; The following objects are masked from 'package:SingleR':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     BlueprintEncodeData, DatabaseImmuneCellExpressionData,</span>
<span class="co">#&gt;     HumanPrimaryCellAtlasData, ImmGenData, MonacoImmuneData,</span>
<span class="co">#&gt;     MouseRNAseqData, NovershternHematopoieticData</span>

<span class="va">ref_monaco</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/datasets.html">MonacoImmuneData</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; snapshotDate(): 2020-12-03</span>
<span class="co">#&gt; see ?celldex and browseVignettes('celldex') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; see ?celldex and browseVignettes('celldex') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>

<span class="va">pred_monaco_main</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html">SingleR</a></span><span class="op">(</span>test <span class="op">=</span> <span class="va">sce</span>, ref <span class="op">=</span> <span class="va">ref_monaco</span>, labels <span class="op">=</span> <span class="va">ref_monaco</span><span class="op">$</span><span class="va">label.main</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">pred_monaco_main</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;         B cells    CD4+ T cells    CD8+ T cells Dendritic cells       Monocytes </span>
<span class="co">#&gt;             348             894             294              42             632 </span>
<span class="co">#&gt;        NK cells     Progenitors         T cells </span>
<span class="co">#&gt;             160              15             258</span>
<span class="va">sce</span><span class="op">$</span><span class="va">labels_main</span> <span class="op">&lt;-</span> <span class="va">pred_monaco_main</span><span class="op">$</span><span class="va">labels</span>
<span class="va">pred_monaco_fine</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html">SingleR</a></span><span class="op">(</span>test <span class="op">=</span> <span class="va">sce</span>, ref <span class="op">=</span> <span class="va">ref_monaco</span>, labels <span class="op">=</span> <span class="va">ref_monaco</span><span class="op">$</span><span class="va">label.fine</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">pred_monaco_fine</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    Central memory CD8 T cells           Classical monocytes </span>
<span class="co">#&gt;                            63                           320 </span>
<span class="co">#&gt;   Effector memory CD8 T cells             Exhausted B cells </span>
<span class="co">#&gt;                            40                            41 </span>
<span class="co">#&gt;     Follicular helper T cells        Intermediate monocytes </span>
<span class="co">#&gt;                           130                           243 </span>
<span class="co">#&gt;                    MAIT cells       Myeloid dendritic cells </span>
<span class="co">#&gt;                            79                            42 </span>
<span class="co">#&gt;                 Naive B cells             Naive CD4 T cells </span>
<span class="co">#&gt;                           202                           250 </span>
<span class="co">#&gt;             Naive CD8 T cells          Natural killer cells </span>
<span class="co">#&gt;                            73                           150 </span>
<span class="co">#&gt;       Non classical monocytes   Non-switched memory B cells </span>
<span class="co">#&gt;                            64                            70 </span>
<span class="co">#&gt;            Non-Vd2 gd T cells                  Plasmablasts </span>
<span class="co">#&gt;                            16                             6 </span>
<span class="co">#&gt;  Plasmacytoid dendritic cells              Progenitor cells </span>
<span class="co">#&gt;                             4                            14 </span>
<span class="co">#&gt;       Switched memory B cells            T regulatory cells </span>
<span class="co">#&gt;                            29                            95 </span>
<span class="co">#&gt; Terminal effector CD4 T cells Terminal effector CD8 T cells </span>
<span class="co">#&gt;                             4                            65 </span>
<span class="co">#&gt;                     Th1 cells                Th1/Th17 cells </span>
<span class="co">#&gt;                           144                           125 </span>
<span class="co">#&gt;                    Th17 cells                     Th2 cells </span>
<span class="co">#&gt;                            65                           179 </span>
<span class="co">#&gt;                Vd2 gd T cells </span>
<span class="co">#&gt;                           130</span>
<span class="va">sce</span><span class="op">$</span><span class="va">labels_fine</span> <span class="op">&lt;-</span> <span class="va">pred_monaco_fine</span><span class="op">$</span><span class="va">labels</span>
<span class="va">pred_monaco_ont</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html">SingleR</a></span><span class="op">(</span>test <span class="op">=</span> <span class="va">sce</span>, ref <span class="op">=</span> <span class="va">ref_monaco</span>, labels <span class="op">=</span> <span class="va">ref_monaco</span><span class="op">$</span><span class="va">label.ont</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">pred_monaco_ont</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; CL:0000236 CL:0000545 CL:0000546 CL:0000623 CL:0000782 CL:0000784 CL:0000788 </span>
<span class="co">#&gt;         43        130        185        154         42          4        201 </span>
<span class="co">#&gt; CL:0000798 CL:0000815 CL:0000860 CL:0000875 CL:0000895 CL:0000899 CL:0000900 </span>
<span class="co">#&gt;        108         99        319         64        235         58         80 </span>
<span class="co">#&gt; CL:0000907 CL:0000912 CL:0000913 CL:0000940 CL:0000970 CL:0000972 CL:0000980 </span>
<span class="co">#&gt;         70        131         62         86         69         28          6 </span>
<span class="co">#&gt; CL:0001044 CL:0001062 CL:0002038 CL:0002043 CL:0002393 </span>
<span class="co">#&gt;          3         63        143         13        247</span>
<span class="va">sce</span><span class="op">$</span><span class="va">labels_ont</span> <span class="op">&lt;-</span> <span class="va">pred_monaco_ont</span><span class="op">$</span><span class="va">labels</span>

<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">labels_main</span>,
      <span class="va">sce</span><span class="op">$</span><span class="va">labels_fine</span><span class="op">)</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Central memory CD8 T cells Classical monocytes</span>
<span class="co">#&gt;   B cells                                  0                   0</span>
<span class="co">#&gt;   CD4+ T cells                             5                   0</span>
<span class="co">#&gt;   CD8+ T cells                            49                   0</span>
<span class="co">#&gt;   Dendritic cells                          0                   4</span>
<span class="co">#&gt;   Monocytes                                0                 315</span>
<span class="co">#&gt;   NK cells                                 0                   0</span>
<span class="co">#&gt;   Progenitors                              0                   1</span>
<span class="co">#&gt;   T cells                                  9                   0</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Effector memory CD8 T cells Exhausted B cells</span>
<span class="co">#&gt;   B cells                                   0                41</span>
<span class="co">#&gt;   CD4+ T cells                              0                 0</span>
<span class="co">#&gt;   CD8+ T cells                             29                 0</span>
<span class="co">#&gt;   Dendritic cells                           0                 0</span>
<span class="co">#&gt;   Monocytes                                 0                 0</span>
<span class="co">#&gt;   NK cells                                  0                 0</span>
<span class="co">#&gt;   Progenitors                               0                 0</span>
<span class="co">#&gt;   T cells                                  11                 0</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Follicular helper T cells Intermediate monocytes MAIT cells</span>
<span class="co">#&gt;   B cells                                 0                      0          0</span>
<span class="co">#&gt;   CD4+ T cells                          125                      0          1</span>
<span class="co">#&gt;   CD8+ T cells                            4                      0          2</span>
<span class="co">#&gt;   Dendritic cells                         0                      1          0</span>
<span class="co">#&gt;   Monocytes                               0                    242          0</span>
<span class="co">#&gt;   NK cells                                0                      0          0</span>
<span class="co">#&gt;   Progenitors                             0                      0          0</span>
<span class="co">#&gt;   T cells                                 1                      0         76</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Myeloid dendritic cells Naive B cells Naive CD4 T cells</span>
<span class="co">#&gt;   B cells                               0           200                 0</span>
<span class="co">#&gt;   CD4+ T cells                          0             0               200</span>
<span class="co">#&gt;   CD8+ T cells                          0             0                46</span>
<span class="co">#&gt;   Dendritic cells                      33             0                 0</span>
<span class="co">#&gt;   Monocytes                             9             1                 0</span>
<span class="co">#&gt;   NK cells                              0             0                 0</span>
<span class="co">#&gt;   Progenitors                           0             0                 0</span>
<span class="co">#&gt;   T cells                               0             1                 4</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Naive CD8 T cells Natural killer cells</span>
<span class="co">#&gt;   B cells                         0                    0</span>
<span class="co">#&gt;   CD4+ T cells                    9                    0</span>
<span class="co">#&gt;   CD8+ T cells                   63                    0</span>
<span class="co">#&gt;   Dendritic cells                 0                    0</span>
<span class="co">#&gt;   Monocytes                       0                    0</span>
<span class="co">#&gt;   NK cells                        0                  148</span>
<span class="co">#&gt;   Progenitors                     0                    0</span>
<span class="co">#&gt;   T cells                         1                    2</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Non classical monocytes Non-switched memory B cells</span>
<span class="co">#&gt;   B cells                               0                          70</span>
<span class="co">#&gt;   CD4+ T cells                          0                           0</span>
<span class="co">#&gt;   CD8+ T cells                          0                           0</span>
<span class="co">#&gt;   Dendritic cells                       0                           0</span>
<span class="co">#&gt;   Monocytes                            64                           0</span>
<span class="co">#&gt;   NK cells                              0                           0</span>
<span class="co">#&gt;   Progenitors                           0                           0</span>
<span class="co">#&gt;   T cells                               0                           0</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Non-Vd2 gd T cells Plasmablasts Plasmacytoid dendritic cells</span>
<span class="co">#&gt;   B cells                          0            6                            0</span>
<span class="co">#&gt;   CD4+ T cells                     0            0                            0</span>
<span class="co">#&gt;   CD8+ T cells                     3            0                            0</span>
<span class="co">#&gt;   Dendritic cells                  0            0                            4</span>
<span class="co">#&gt;   Monocytes                        0            0                            0</span>
<span class="co">#&gt;   NK cells                         4            0                            0</span>
<span class="co">#&gt;   Progenitors                      0            0                            0</span>
<span class="co">#&gt;   T cells                          9            0                            0</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Progenitor cells Switched memory B cells T regulatory cells</span>
<span class="co">#&gt;   B cells                        0                      29                  0</span>
<span class="co">#&gt;   CD4+ T cells                   0                       0                 90</span>
<span class="co">#&gt;   CD8+ T cells                   0                       0                  4</span>
<span class="co">#&gt;   Dendritic cells                0                       0                  0</span>
<span class="co">#&gt;   Monocytes                      0                       0                  0</span>
<span class="co">#&gt;   NK cells                       0                       0                  0</span>
<span class="co">#&gt;   Progenitors                   14                       0                  0</span>
<span class="co">#&gt;   T cells                        0                       0                  1</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Terminal effector CD4 T cells Terminal effector CD8 T cells</span>
<span class="co">#&gt;   B cells                                     0                             0</span>
<span class="co">#&gt;   CD4+ T cells                                0                             0</span>
<span class="co">#&gt;   CD8+ T cells                                2                            37</span>
<span class="co">#&gt;   Dendritic cells                             0                             0</span>
<span class="co">#&gt;   Monocytes                                   0                             0</span>
<span class="co">#&gt;   NK cells                                    0                             5</span>
<span class="co">#&gt;   Progenitors                                 0                             0</span>
<span class="co">#&gt;   T cells                                     2                            23</span>
<span class="co">#&gt;                  </span>
<span class="co">#&gt;                   Th1 cells Th1/Th17 cells Th17 cells Th2 cells Vd2 gd T cells</span>
<span class="co">#&gt;   B cells                 0              1          0         0              1</span>
<span class="co">#&gt;   CD4+ T cells          122            104         62       172              4</span>
<span class="co">#&gt;   CD8+ T cells           15              3          1         7             29</span>
<span class="co">#&gt;   Dendritic cells         0              0          0         0              0</span>
<span class="co">#&gt;   Monocytes               0              0          0         0              1</span>
<span class="co">#&gt;   NK cells                0              0          0         0              3</span>
<span class="co">#&gt;   Progenitors             0              0          0         0              0</span>
<span class="co">#&gt;   T cells                 7             17          2         0             92</span></code></pre></div>
<p>This concludes the preparation of the data. We have now a <code>SingleCellExperiment</code> object that contains different types of abundance values, representations in reduced dimensions, as well as a range of row (feature) and column (cell) metadata.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://iSEE.github.io/iSEEWorkshopEuroBioc2020">iSEEWorkshopEuroBioc2020</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/load_demo_sce.html">cache_demo_sce</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"pbmc3k"</span><span class="op">)</span>
<span class="co">#&gt; Saving object to /github/home/.cache/iSEEWorkshopEuroBioc2020/pbmc3k.rds</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kevin Rue-Albrecht, Federico Marini, Charlotte Soneson, Aaron Lun.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
